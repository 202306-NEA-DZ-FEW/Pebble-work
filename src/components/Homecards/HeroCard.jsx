import React from "react";
import { useTranslation } from "next-i18next";
import styles from "@/styles/Homepage.module.css";

function HeroCard() {
    const { t } = useTranslation();

    const handleButtonClick = (path) => {
        window.location.href = path;
    };

    return (
        <div className='w-full shadow-md sm:h-[400px] flex justify-center items-center'>
            <div className='text-center p-4'>
                <div className='max-w-screen-xl mx-auto'>
                    <div className='text-4xl font-medium leading-10 tracking-tight text-zinc-900'>
                        {t("common:heroCard:title")}
                    </div>
                    <div className='text-xl font-normal leading-10 tracking-tight text-zinc-500'>
                        {t("common:heroCard:description")}
                    </div>
                    <div className=' flex flex-col sm:flex-row justify-center items-center gap-6'>
                        <div className='cursor-pointer'>
                            <button
                                className={`buttonTransition text-2xl font-medium leading-10 tracking-tight text-white px-4 py-2 rounded-lg`}
                                onClick={() =>
                                    handleButtonClick("/target-page-1")
                                }
                            >
                                {t("common:heroCard:getStarted")}
                            </button>
                        </div>
                        <div className='cursor-pointer'>
                            <div
                                className='text-2xl font-medium leading-10 tracking-tight text-zinc-900'
                                onClick={() =>
                                    handleButtonClick("/target-page-2")
                                }
                            >
                                {t("common:heroCard:whoWeAre")}
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default HeroCard;
