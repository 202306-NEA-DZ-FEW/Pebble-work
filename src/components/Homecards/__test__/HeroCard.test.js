import React from "react";
import { render } from "@testing-library/react";
import HeroCard from "@/components/Homecards/HeroCard";

describe("HeroCard", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<HeroCard />);
        expect(asFragment()).toMatchSnapshot();
    });
});
