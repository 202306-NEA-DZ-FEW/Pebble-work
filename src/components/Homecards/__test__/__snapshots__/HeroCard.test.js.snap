// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full shadow-md sm:h-[400px] flex justify-center items-center"
  >
    <div
      class="text-center p-4"
    >
      <div
        class="max-w-screen-xl mx-auto"
      >
        <div
          class="text-4xl font-medium leading-10 tracking-tight text-zinc-900"
        >
          common:heroCard:title
        </div>
        <div
          class="text-xl font-normal leading-10 tracking-tight text-zinc-500"
        >
          common:heroCard:description
        </div>
        <div
          class=" flex flex-col sm:flex-row justify-center items-center gap-6"
        >
          <a
            class="buttonTransition text-[22px] font-medium leading-10 tracking-tight text-white px-4 py-2 rounded-lg"
            href="/signup"
          >
            common:heroCard:getStarted
          </a>
          <a
            class="secondButton text-[2xl] font-medium leading-10 tracking-tight text-zinc-900"
            href="/guidelines"
          >
            common:heroCard:whoWeAre
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
